alica_test_utility/include/alica/test/SuccessOrFailBehaviour.hpp

```diff
@@ -0,0 +1,83 @@
+#pragma once
+
+#include <engine/BasicBehaviour.h>
+
+namespace alica::test
+{
+template <uint32_t iterationsBeforeTrigger>
+class SuccessOrFailBehaviour : public alica::BasicBehaviour
+{
+public:
+    SuccessOrFailBehaviour<iterationsBeforeTrigger>();
+    void run(void* msg) override;
+    void setTriggerType(bool triggerSuccess);
+
+protected:
+    uint32_t iterationsCounter() const;
+    void incIterationsCounter();
+    void initialiseParameters() override;
+
+private:
+    std::mutex _triggerMutex;
+    bool _triggerTypeSet;
+    bool _triggerSuccess;
+    uint32_t _iterationsBeforeTrigger;
+    uint32_t _iterationsCounter;
+};
+
+template <uint32_t iterationsBeforeTrigger>
+SuccessOrFailBehaviour<iterationsBeforeTrigger>::SuccessOrFailBehaviour()
+        : BasicBehaviour("SuccessOrFailBehaviour")
+        , _iterationsCounter(0)
+        , _iterationsBeforeTrigger(iterationsBeforeTrigger)
+        , _triggerTypeSet(false)
+        , _triggerSuccess(false)
+{
+}
+
+template <uint32_t iterationsBeforeTrigger>
+uint32_t SuccessOrFailBehaviour<iterationsBeforeTrigger>::iterationsCounter() const
+{
+    return _iterationsCounter;
+}
+
+template <uint32_t iterationsBeforeTrigger>
+void SuccessOrFailBehaviour<iterationsBeforeTrigger>::incIterationsCounter()
+{
+    _iterationsCounter++;
+}
+
+template <uint32_t iterationsBeforeTrigger>
+void SuccessOrFailBehaviour<iterationsBeforeTrigger>::setTriggerType(bool triggerSuccess)
+{
+    std::lock_guard<std::mutex> lockGuard(_triggerMutex);
+    _triggerSuccess = triggerSuccess;
+    _triggerTypeSet = true;
```

Comment:
Can we take this is the constructor?

Reply:
Theoretical yes, but currently we require the mockup behaviours to be default-constructable, in order to keep the complexity of the templates of the TestBehaviourGenerator simple.